yaptide/utils/sim_utils.py

```diff
@@ -33,19 +36,90 @@ def pymchelper_output_to_json(estimators_dict: dict, dir_path: Path) -> dict:
+
+    Editor = auto()
+    Files = auto()
+
+
+def get_json_type(payload_dict: dict) -> JSON_TYPE:
+    """Returns type of provided JSON"""
+    possible_input_file_names = set(['beam.dat', 'geo.dat', 'detect.dat', 'mat.dat'])  # skipcq: PTC-W0018
+    if possible_input_file_names.intersection(set(payload_dict["sim_data"].keys())):
+        return JSON_TYPE.Files
+    return JSON_TYPE.Editor
+
+
+def convert_editor_payload_to_dict(editor_dict: dict, parser_type: str) -> dict:
+    """
+    Convert payload data to dictionary with filenames and contents for Editor type projects
+    Otherwise return empty dictionary
+    """
+    conv_parser = get_parser_from_str(parser_type)
+    files_dict = run_parser(parser=conv_parser, input_data=editor_dict)
+    return files_dict
+
+
+def check_and_convert_payload_to_files_dict(payload_dict: dict) -> dict:
     """
-    Function used to write input files to output directory.
-    Returns dictionary with filenames as keys and their content as values
+    Convert payload data to dictionary with filenames and contents for Editor type projects
+    Otherwise return empty dictionary
     """
-    if "beam.dat" not in json_data["sim_data"]:
-        conv_parser = get_parser_from_str(json_data["sim_type"])
-        return run_parser(parser=conv_parser, input_data=json_data["sim_data"], output_dir=output_dir)
-
-    for key, file in json_data["sim_data"].items():
-        with open(Path(output_dir, key), "w") as writer:
-            writer.write(file)
-    return json_data["sim_data"]
+    files_dict = {}
+    json_type = get_json_type(payload_dict)
+    if json_type == JSON_TYPE.Editor:
+        files_dict = convert_editor_payload_to_dict(editor_dict=payload_dict["sim_data"],
+                                                    parser_type=payload_dict["sim_type"])
+    else:
+        logging.warning("Project of %s used, conversion works only for Editor projects", json_type)
+    return files_dict
+
+
+def editor_dict_with_adjusted_primaries(payload_editor_dict: dict) -> dict:
+    """Replace number of primaries in editor dict"""
+    editor_dict = copy.deepcopy(payload_editor_dict['sim_data'])
+    editor_dict['beam']['numberOfParticles'] //= payload_editor_dict['ntasks']
+    return editor_dict
+
+
+def files_dict_with_adjusted_primaries(payload_files_dict: dict) -> dict:
+    """Replace number of primaries in files dict"""
+    files_dict = copy.deepcopy(payload_files_dict['sim_data'])
+    all_beam_lines: list[str] = files_dict['beam.dat'].split('\n')
+    all_beam_strings_with_nstat = [line for line in all_beam_lines if 'NSTAT' in line]
+    if len(all_beam_strings_with_nstat) != 1:
+        return files_dict
+    old_nstat: str = all_beam_strings_with_nstat[0].split()[1]
+    new_nstat = str(int(old_nstat) // payload_files_dict['ntasks'])
+    for i in range(len(all_beam_lines)):
+        if 'NSTAT' in all_beam_lines[i]:
+            all_beam_lines[i] = all_beam_lines[i].replace(old_nstat, new_nstat, 1)
```

Comment:
Lets imagine that you have following line:
`NSTAT 100 1000 ! bla bla 100 10000`
and that ntasks = 1
After running this `replace` you will get:
`NSTAT 10 100 ! bla bla 10 1000`

That is not correct !

Read documentation about NSTAT, it has two arguments if I remember correctly.

Also - write good tests for this method.

Another idea -> extract the code which replaces single line with NSTAT to a separate method and write good tests for it

Reply:
You are wrong
Example line `NSTAT 100 1000 ! bla bla 100 10000` will be converted to `NSTAT 10 1000 ! bla bla 100 10000`
3rd argument of `replace` method specifies the count of matched occurances to be replaced.

Reply:
```suggestion
            # replace first occurrence of the number of particles, going through the string from the left
            all_beam_lines[i] = all_beam_lines[i].replace(old=old_nstat, new=new_nstat, count=1)
```

Correct, then I suggest to increase code readability

Reply:
This doesn't work because: `TypeError: str.replace() takes no keyword arguments`
I will add proper comment

Reply:
Correct, comment should be enough